docs/_content/themas/achtergronddocumentatie/versies-en-migraties.md

```diff
@@ -16,24 +16,28 @@ Versiebeheer is gebaseerd op [Semantic Versioning](https://semver.org) en de
 Elke  API heeft een eigen versienummer, opgebouwd uit `MAJOR.MINOR.PATCH` 
 element. Bijvoorbeeld versie `2.1.8`:
 
-Zie: [VNG API-Beheer](https://github.com/VNG-Realisatie/api-beheer/blob/master/versiebeheer.md)
+In [VNG API-Beheer](https://github.com/VNG-Realisatie/api-beheer/blob/master/versiebeheer.md) kun je lezen dat VNG API Beheer maximaal 2 major versies van elke API standaard ondersteund. Dat staat echter los van het aantal versies van een API dat door een provider applicatie wordt ondersteund. Zo staat het een leverancier vrij om gelijktijdig alle ooit beschikbaar gestelde API's te ondersteunen. In de onderstaande scenario's gaan we daar niet vanuit.
 
```

Comment:
Inmiddels hebben we last van voortschrijdend inzicht en betwijfel ik of we nog wel zoveel versies moeten gaan ondersteunen. Uiteindelijk is het doel om leveranciers en gemeenten te stimuleren mee te gaan met de ontwikkelingen en meerdere kleine stapjes die snel inspelen op de behoeften van dat moment lijken beter aan te sluiten bij de wensen van leveranciers en gemeenten dan een paar keer per jaar grotere wijzigingen.

Reply:
Wat bedoel je met 'zoveel'? 2 major versies is toch niet zo veel en lijkt me het minimale.
Ik ben het overigens wel met je eens dat we beter veel kleine stapjes kunnen nemen zodat gemeenten en leveranciers het kunnen bijbenen.
Ook majors moeten denk ik zo klein mogelijke stapjes nemen met hetzelfde doel. Dat zou echter kunnen betekenen dat je toch meer dan 2 majors per jaar uitbrengt.

Reply:
Precies dat vind ik lastig: wanneer we meer dan 2 majors per jaar uitbrengen en alleen de laatste 2 majors ondersteunen is een versie wel heel vluchtig. Waarom zouden we dan die laatste twee wel ondersteunen? Dan is alleen de laatste versie ondersteunen net zo valide te noemen.

Reply:
Misschien moeten we het gewoon anders definiëren.
We kunnen een tijdsbestek afspreken waarbinnen een oude major versie nog ondersteund blijft door ons. Dat tijdsbestek moet dan zo gekozen worden dat het voor een ontwikkelaar doenlijk is om daarbinnen zijn software aan te passen.
Dat zou kunnen betekenen dat we soms meer dan 2 versies, soms 2 versies maar misschien ook soms maar 1 versie hoeven te ondersteunen.
De vraag is nog hoe het in de praktijk uitpakt, dat moeten we denk ik gewoon gaan zien. Wellicht valt het allemaal wel mee. We kunnen deze vraag misschien ook eens stellen aan anderen.

Reply:
Ik zou het ook niet in beton gieten. Maar aangezien leveranciers enige zekerheid willen kan je bijv. aangeven:

1. Maximaal 2 major versies per jaar
2. Minimaal 2 jaar ondersteuning op elke gereleaste versie (laatste minor.patch van een major)

Worst case zit je na 2 jaar op 4 ondersteunde versies. Best case met 1 versie.

Wat dan relevante vragen zijn: Wat betekent dit voor leveranciers? Gemeentes? En Beheer?

Impact voor beheer kunnen jullie het beste inschatten maar lijkt me te overzien.

Aangezien REST APIs inherent veel eenvoudiger zijn, zal ondersteuning voor nieuwe versies veel sneller te ontwikkelen zijn voor leveranciers (vermoed ik) mits er goede migratie beschreven wordt. Tezamen met het punt hieronder voor gemeentes zal dit inbouwen toch inherent moeten worden.

Voor gemeentes lijkt het me (mentaal) nog het lastigst. Die moeten af van versies inkopen en toe naar actief versie onderhoud van hun pakketten.

Reply:
De vraag is wat ondersteuning precies betekent.
Zoals het nu beschreven staat betekent dat bijv. dat we aanpassingen op documentatie op alle ondersteunde versies moeten doorvoeren tenzij het documentatie betreft van een aanpassing die niet op een/de oudere versie(s) van toepassing is. Dat zou dus kunnen betekenen dat de documentatie voor 4 verschillende versies moet worden aangepast.
We kunnen ook overwegen om het niveau van ondersteuning van oude major.minors te verlagen. Bijv. wel vragen beantwoorden en hosting voor de Referentie Implementatie bieden maar geen patches meer uitbrengen op oude versies.

Reply:
Dat kan natuurlijk!